src/common/local_file_system.cpp

```diff
@@ -267,6 +267,37 @@ static string AdditionalProcessInfo(FileSystem &fs, pid_t pid) {
 }
 #endif
 
+void LocalFileSystem::CreatePrivateFile(const string &path_p, FileOpener *opener) {
+	auto path = FileSystem::ExpandPath(path_p, opener);
+
+	int open_flags = O_WRONLY;
+	open_flags |= O_CLOEXEC;
+	open_flags |= O_CREAT;
+	open_flags |= O_APPEND;
+
+	int fd = open(path.c_str(), open_flags, 0600);
```

Comment:
Maybe we can pass this as a flag to `OpenFile` instead of having this be a separate API call?

Reply:
Yep good idea